pkg/cluster/cluster.go

```diff
@@ -409,6 +409,23 @@ func createClusterSpec(config Spec, awsClient aws.Client) (*cmv1.Cluster, error)
 		}
 	}
 
+	if config.AvailabilityZones != nil {
+		clusterBuilder = clusterBuilder.Nodes(
+			cmv1.NewClusterNodes().AvailabilityZones(config.AvailabilityZones...),
```

Comment:
I think this might override the existing Nodes object already built a few blocks above.

Reply:
> I think this might override the existing Nodes object already built a few blocks above.

> Since rosa is heavily integrated with the AWS SDK, and we have a session open already during cluster creation, it should be fairly easy to query for a list of availabilty zones and present those as a MultiSelect prompt. Having selected AZs, then we can query AWS for the subnet IDs and have a new prompt for the user to select those as well

I guess we can do that - the main drawback is that **when** the user created her subnets aws has created them in _specific_ az's (we have preflight to verify that the subnets are in the az's the user supplied) so in fact its not really a choice for the user; they are simply filling in the az's from the aws console. I think that perhaps it could be easier UX if we do show them a list rather ask them to fill that in but am not sure - WDYT?

In that case we can also query for the subnets as well..

Reply:
Can we just show a list of subnets for the `region` and let the tool auto-fill the AZs without user interaction?

Reply:
Ummm no ..

Reply:
You can do something like:
```
subnetList = aws.ec2client.DescribeSubnets() // fetch subnets from AWS
subnets = interactive.GetMultiOptions(subnetList) // let user select subnets
foreach subnet in subnets {
  foreach subnetObj in subnetList {
    if subnet == subnetObj.id {
      az = append(az, subnetObj.AvailabilityZone) // create an internal list of matching AZs
    }
  }
}
createCluster({subnets: subnets, availabilityZones: az}) // send to cluster creation
```

Reply:
> Ummm no ..

Why not? Do we need the user to select the AZs explicitly?

Reply:
Alternatively we can do something like:
```
azList = aws.ec2client.DescribeAvailabilityZones(region)
azs = interactive.GetMultiOptions(azList)

subnetList = aws.ec2client.DescribeSubnets(region, azs)
subnets = interactive.GetMultiOptions(subnetList)

createCluster(subnets, azs)
```
that way the user selects AZs first and we show the list of subnets from the AZs they've already selected

Reply:
@vkareh I looked that thsubnet list and I think one can even do better
```
subnetList = aws.ec2client.DescribeSubnets(region, azs)
subnets = interactive.GetMultiOptions(subnetList)
for subnet := range subnets.Items {
    availabilityZones = append(availabilityZones, aws.String(subnet.AvailabilityZone))
}

createCluster(subnets, availabilityZones)
```

Reply:
Oh its essentially youre first options :facepalm:

Reply:
haha we're on the same page! good! :)

Reply:
@vkareh Using a `map` I was able to optimize the code a little bit, please take a look.